torch/csrc/jit/python/init.cpp

```diff
@@ -1699,8 +1699,15 @@ void initJITBindings(PyObject* module) {
           [](SchemaInfo& self,
              const std::string& name,
              const py::object& value) {
+            // For normalization purposes there is an inconsistency within
+            // torch.fx that turns all arguments named "self" into "input". Thus
+            // this check ensures that those arguments are checked correctly.
             if (name == "input") {
-              self.addArgumentValue("self", toTypeInferredIValue(value));
+              try {
+                self.addArgumentValue("self", toTypeInferredIValue(value));
+              } catch (const c10::Error& e) {
+                self.addArgumentValue("input", toTypeInferredIValue(value));
+              }
```

Comment:
nit: would it be possible to instead add a `hasArgumentNamed(std::string)` or similar into `SchemaInfo` so this can be handled with an if/else instead of with exceptions?